plasma_framework/contracts/src/fast_exits/liquidity.sol

```diff
@@ -0,0 +1,202 @@
+
+            FungibleTokenOutputModel.Output memory outputFromSecondTransaction
+         = decodedSecondTx.outputs[0];
+        uint256 amount = outputFromSecondTransaction.amount;
+        exitIdtoAmount[_exitId] = amount;
+    }
+
+    /**
+     * @notice Check if the person calling is the same person who created the tx to the contract
+     * @param _rlpInputCreationTx RLP-encoded first transaction that transfers to this contract
+     * @param _utxoPosInput position of the output that created the inputs for second transaction
+    */
+    function verifyOwnership(
+        bytes memory _rlpInputCreationTx,
+        uint256 _utxoPosInput
+    ) internal {
+
+            PaymentTransactionModel.Transaction memory decodedFirstTx
+         = PaymentTransactionModel.decode(_rlpInputCreationTx);
+        uint16 firstTransactionOutputIndex = getOutputIndex(_utxoPosInput);
+
+            FungibleTokenOutputModel.Output memory outputFromFirstTransaction
+         = decodedFirstTx.outputs[firstTransactionOutputIndex];
+        address ownerFirstTxOutput = address(
+            uint160(outputFromFirstTransaction.outputGuard)
+        );
+        require(
+            ownerFirstTxOutput == msg.sender,
+            "First Tx Provided was not called by sender"
+        );
+    }
+
+    /**
+     * @notice Verify the First Tx provided is valid
+     * @param _utxoPosInput position of the output that created the inputs for second transaction
+     * @param _rlpInputCreationTx RLP-encoded first transaction that transfers to this contract
+     * @param _inputCreationTxInclusionProof First transactions inclusion proofs
+    */
+    function verifyTxValidity(
+        uint256 _utxoPosInput,
+        bytes memory _rlpInputCreationTx,
+        bytes memory _inputCreationTxInclusionProof
+    ) internal {
+        PosLib.Position memory utxoDecoded = PosLib.decode(_utxoPosInput);
+        utxoDecoded.outputIndex = 0;
+        (bytes32 _root, ) = pf.blocks(utxoDecoded.blockNum);
+        require(_root != bytes32(""), "Failed to get root of the block");
+        bool txExists = Merkle.checkMembership(
+            _rlpInputCreationTx,
+            utxoDecoded.txIndex,
+            _root,
+            _inputCreationTxInclusionProof
+        );
+        require(
+            txExists,
+            "Provided Transaction isn't finalized or doesn't exist"
+        );
+    }
+
+    /**
+     * @notice func that calls omg-contracts to start the exit
+     * @param _utxoPosToExit position of the output which the contract has to exit
+     * @param _rlpOutputTxToContract RLP-encoded transaction that creates the output for the contract
+     * @param _outputTxToContractInclusionProof Second Transaction's inclusion proof
+    */
+    function runExit(
+        uint256 _utxoPosToExit,
+        bytes memory _rlpOutputTxToContract,
+        bytes memory _outputTxToContractInclusionProof
+    ) internal {
+        PaymentStandardExitRouterArgs.StartStandardExitArgs memory s;
```

Comment:
minor: slight preference of using the following style instead:
```
PaymentStandardExitRouterArgs.StartStandardExitArgs({
    utxoPos: _utxoPosToExit,
    rlpOutputTx: _rlpOutputTxToContract,
    ....: ...
})
```

reason is that this style would force you to provide information for all fields and avoid some manual mistakes.